src/CloudFoundry.VisualStudio/MSBuild/MSBuildProcess.cs

```diff
@@ -2,22 +2,23 @@
 {
     using System;
     using System.Collections.Generic;
+    using System.IO;
     using System.Linq;
     using System.Text;
     using System.Threading.Tasks;
     using EnvDTE;
+    using EnvDTE80;
     using Microsoft.Build.Evaluation;
-    using Microsoft.VisualStudio.Shell;
-    using Microsoft.Build.Framework;
     using Microsoft.Build.Execution;
-    using Microsoft.VisualStudio.Shell.Interop;
-    using Microsoft.VisualStudio.ComponentModelHost;
+    using Microsoft.Build.Framework;
     using Microsoft.VisualStudio;
-    using EnvDTE80;
-    using System.IO;
-
+    using Microsoft.VisualStudio.ComponentModelHost;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Shell.Interop;
+    
     public class MSBuildProcess
     {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly", Justification = "Build properties are set on finish button click")]
```

Comment:
That is not a good enough reason to make the property settable.
If the collection needs to be replaced, clear it, then add the new items.